server/services/voip/service.ts

```diff
@@ -143,35 +143,42 @@ export class VoipService extends ServiceClass implements IVoipService {
 		return this.commandHandler.executeCommand(Commands.extension_list, undefined);
 	}
 
-	async getExtensionDetails(requestParams: any): Promise<IVoipConnectorResult> {
+	async getExtensionDetails(requestParams: any): Promise<IVoipConnectorResult> { // TODO remove any
 		return this.commandHandler.executeCommand(
 			Commands.extension_info,
 			requestParams);
 	}
 
-	async getRegistrationInfo(requestParams: any): Promise<IVoipConnectorResult> {
-		const config: IVoipServerConfig = Promise.await(
-			this.getServerConfigData(ServerType.CALL_SERVER),
-		) as unknown as IVoipServerConfig;
+	async getRegistrationInfo(requestParams: { extension: string }): Promise<{result: IRegistrationInfo}> {
+		const config = await this.getServerConfigData(ServerType.CALL_SERVER);
 
 		if (!config) {
 			this.logger.warn({ msg: 'API = connector.extension.getRegistrationInfo callserver settings not found' });
-			return Promise.reject('Not found');
+			throw new Error('Not found');
 		}
 
-		const endpointDetails = Promise.await(this.commandHandler.executeCommand(
+		const endpointDetails = await this.commandHandler.executeCommand(
 			Commands.extension_info,
 			requestParams,
-		)) as IVoipConnectorResult;
-		const callServerConfig: ICallServerConfigData = config.configData as ICallServerConfigData;
-		const extensionDetails: IExtensionDetails = endpointDetails.result as unknown as IExtensionDetails;
+		);
+		if (!isIExtensionDetails(endpointDetails.result)) { // TODO The result and the assertion doenst match amol please check
```

Comment:
Should I approve this without Amol check? Why?

Reply:
We can always discuss this later

Reply:
don't forget to mention that to Amol...